### ISSUE TITLE
Crashes on startup if beacon node not yet ready.

### ISSUE THREAD

**Issue by timmyers** [Alethio/eth2stats-client#19]
It seems during the start-up, if the beacon node is not yet ready to serve traffic, `eth2stats` will crash.

This causes a problem when running both of them in a kubernetes pod together.  The pod starts up, starting a beacon node, and `eth2stats`.  The beacon node takes a little time to be able to start serving traffic, but `eth2stats` attempts immediately, fails, and crashes.  This causes all the containers to be restarted, and we get into an infinite loop of failure.

Would it make sense to have some retry/back-off mechanism at startup to be more forgiving of the beacon node not yet being ready?

**Comment by tzapu** [comment]
weird, it was supposed to wait indefinitely.

thanks, will investigate.

does it give any specific error message when crashing?

**Comment by timmyers** [comment]
Not really, usually I am just seeing logs stop with no error other than RPC failures.  The full output will generally be one of these two cases:

```
time="2020-02-10T23:57:26Z" level=info msg="Could not load config file. Falling back to args. Error: Config File \"config\" Not Found in \"[/]\"" module=main
time="2020-02-10T23:57:26Z" level=debug msg="[main] Debug mode"
time="2020-02-10T23:57:26Z" level=info msg="[prysm] setting up beacon client connection"
time="2020-02-10T23:57:26Z" level=info msg="[core] setting up eth2stats server connection"
time="2020-02-10T23:57:26Z" level=debug msg="[core] looking for existing token"
time="2020-02-10T23:57:27Z" level=debug msg="[core] found token"
time="2020-02-10T23:57:27Z" level=info msg="[core] getting beacon client version"
```
OR 
```
time="2020-02-10T23:57:26Z" level=info msg="Could not load config file. Falling back to args. Error: Config File \"config\" Not Found in \"[/]\"" module=main
time="2020-02-10T23:57:26Z" level=debug msg="[main] Debug mode"
time="2020-02-10T23:57:26Z" level=info msg="[prysm] setting up beacon client connection"
time="2020-02-10T23:57:26Z" level=info msg="[core] setting up eth2stats server connection"
time="2020-02-10T23:57:26Z" level=debug msg="[core] looking for existing token"
time="2020-02-10T23:57:27Z" level=debug msg="[core] found token"
time="2020-02-10T23:57:27Z" level=info msg="[core] getting beacon client version"
time="2020-02-10T23:57:27Z" level=info msg="[core] got beacon client version" version="Prysm/Git commit: {STABLE_GIT_COMMIT}. Built at: 2020-02-10T23:39:53Z"
time="2020-02-10T23:57:27Z" level=info msg="[core] getting beacon client genesis time"
time="2020-02-10T23:57:27Z" level=info msg="[core] got beacon client genesis time" genesisTime=1578614400
time="2020-02-10T23:57:27Z" level=info msg="[core] awaiting connection to eth2stats server"
time="2020-02-10T23:57:27Z" level=info msg="[core] getting chain head for initial feed"
time="2020-02-10T23:57:27Z" level=error msg="[prysm] rpc error: code = Internal desc = Could not get finalized block"
time="2020-02-10T23:57:27Z" level=fatal msg="[core] rpc error: code = Internal desc = Could not get finalized block"
```
and then exit with code 1.

**Comment by tzapu** [comment]
thanks, thinking how to handle this. not that trivial because i'm not sure what to use as a "healthy" prysm trigger...

**Comment by tzapu** [comment]
@timmyers could you try with docker image `alethio/eth2stats-client:dev` ?
it should be a lot more forgiving with the kind of interruptions you are worried about
(could be a bit verbose)
would appreciate you testing it as well
cheers

**Comment by timmyers** [comment]
Just tried it out, seemed to work great!

**Comment by tzapu** [comment]
super cool, released it under latest and v0.0.6
